### ISSUE TITLE
Search with GMail returns bad command

### ISSUE THREAD

**Issue by ferhatelmas** [emersion/go-imap#73]
Very simple search with `Before` or `Since` returns bad command error from GMail servers. 

```
oaitgQ SEARCH CHARSET UTF-8 BEFORE " 1-Oct-2016" FROM me@example.com
oaitgQ BAD Could not parse command
```

Any idea to fix ?

**Comment by emersion** [comment]
It looks like Gmail doesn't accept dates with a leading space.

**Comment by ferhatelmas** [comment]
Do you want me to prepare a PR for fix ?

**Comment by emersion** [comment]
Thanks for proposing :P

Should be fixed by 2b50239f40874f92ca308d8b9a83d54e01bb9b1a, can you check?

**Comment by ferhatelmas** [comment]
Thank you too, it works as expected.
